lib/src/install/osconfig.rs

```diff
@@ -40,6 +56,35 @@ fn test_inject_root_ssh() -> Result<()> {
 
     // The code expects this to exist, reasonably so
     root.create_dir("etc")?;
+    // Test with a symlink
+    root.symlink("var/roothome", "root")?;
+    inject_root_ssh_authorized_keys(root, None, "ssh-ed25519 ABCDE example@demo\n").unwrap();
+
+    let content = root.read_to_string(format!("etc/tmpfiles.d/{ROOT_SSH_TMPFILE}"))?;
+    assert_eq!(
+        content,
+        "f~ /var/roothome/.ssh/authorized_keys 600 root root - c3NoLWVkMjU1MTkgQUJDREUgZXhhbXBsZUBkZW1vCg==\n"
+    );
+
+    root.create_dir("etc")?;
```

Comment:
This part of the test is failing right now and seems redundant with the next test, I was inclined to just:
```rust
@@ -60,17 +60,6 @@ fn test_inject_root_ssh() -> Result<()> {
     root.symlink("var/roothome", "root")?;
     inject_root_ssh_authorized_keys(root, None, "ssh-ed25519 ABCDE example@demo\n").unwrap();
 
-    let content = root.read_to_string(format!("etc/tmpfiles.d/{ROOT_SSH_TMPFILE}"))?;
-    assert_eq!(
-        content,
-        "f~ /var/roothome/.ssh/authorized_keys 600 root root - c3NoLWVkMjU1MTkgQUJDREUgZXhhbXBsZUBkZW1vCg==\n"
-    );
-
-    root.create_dir("etc")?;
-    root.remove_file("root")?;
-    root.create_dir("root")?;
-    inject_root_ssh_authorized_keys(root, None, "ssh-ed25519 ABCDE example@demo\n").unwrap();
-
     let content = root.read_to_string(format!("etc/tmpfiles.d/{ROOT_SSH_TMPFILE}"))?;
     assert_eq!(
         content,
```
but then didn't because maybe there is something subtle here that I miss (but I am not seeing it right now :) ?

Reply:
P.S. I wonder if we should setup branch protection for this for "CI / tests (pull_request) ", it seems it does not block the merge right now even with unit tests failing.

Reply:
> P.S. I wonder if we should setup branch protection for this for "CI / tests (pull_request) ", it seems it does not block the merge right now even with unit tests failing.

Yeah I changed the branch rules after dropping Prow but must have messed that up...will look